code/game/objects/items/storage/backpack.dm

```diff
@@ -889,6 +894,25 @@ GLOBAL_LIST_EMPTY_TYPED(radio_packs, /obj/item/storage/backpack/marine/satchel/r
 	fuel_type = "utnapthal"
 	has_gamemode_skin = TRUE
 
+//this is to revert change for the backpack that are for flametrower usage.
+// so that they can use custom mix to refill those backpack
+/obj/item/storage/backpack/marine/engineerpack/flamethrower/afterattack(obj/target, mob/user, proximity)
+	if(!proximity)
+		return
+	if (!(istype(target, /obj/structure/reagent_dispensers/fueltank)))
+		return
+
+	if (reagents.total_volume >= max_fuel)
+		to_chat(user, SPAN_NOTICE("The pack is already full!"))
+		return
+
+	if(reagents.total_volume < max_fuel)
+		target.reagents.trans_to(src, max_fuel)
+		to_chat(user, SPAN_NOTICE("You crack the cap off the top of the pack and fill it back up again from the tank."))
+		playsound(loc, 'sound/effects/refill.ogg', 25, TRUE, 3)
+		return
+	return ..()
```

Comment:
```suggestion
```
Currently impossible to hit this since logic will go either into `if (reagents.total_volume >= max_fuel)` or `if(reagents.total_volume < max_fuel)`